pkgs/test_core/lib/src/runner/vm/test_compiler.dart

```diff
@@ -159,13 +159,23 @@ class _TestCompilerForLanguageVersion {
     final platformDill = 'lib/_internal/vm_platform_strong.dill';
     final sdkRoot =
         p.relative(p.dirname(p.dirname(Platform.resolvedExecutable)));
+    final packageConfigUriAwaited = await packageConfigUri;
+
+    // If we have native assets for the host os in JIT mode, they are here.
+    Uri? nativeAssetsYaml =
+        packageConfigUriAwaited.resolve('native_assets.yaml');
```

Comment:
Should this be a configurable path? Or just wait to allow that until/if it is needed later?

The package config isn't always necessarily what you think it is... tools could copy it to a new location and modify it before invoking us etc.

Reply:
The use case I'm thinking about is `dart test` which is equal to `dart run test:test`. In this case `dart run`/`dart test` will run themselves in JIT mode use the `.dart_tools/package_config.json` and `.dart_tools/native_assets.yaml`.

(`dart test/foo_test.dart` already works after `.dart_tools/package_config.json` and `.dart_tools/native_assets.yaml` are present. Basically running a test that way skips the whole runner infrastructure.)

> tools could [...]

I hope these tools would then still run `dart pub get` + `dart test`/`dart run test` which would then result in the same layout. Are you aware of tools that do different things?

Reply:
> I hope these tools would then still run dart pub get + dart test/dart run test which would then result in the same layout. Are you aware of tools that do different things?

I am not sure if it is still the case but for a long time flutter used to mess with the package config _after_ pub get, to inject some references to generated packages. It did cause problems for build_runner etc or other tools that expect the `.dart_tool/package_config.json` file to be the source of truth, or the `pubspec.yaml` or `pubspec.lock` files to agree with the `package_config.json` file (which was the case with build_runner).

Also as soon as you step outside of the pub ecosystem all these assumptions no longer apply.

Reply:
Yes, for Flutter I have implementations for this in `flutter_tools`. (It does cross-compiling, Dart doesn't. It bundles libraries differently per target OS as well, so it also transforms the paths where libraries are found for the target bundles.)

Reply:
Just a random thought here, but would it make sense to add a new (optional) field to the `package_config.json` which lists the location of the `native_assets.yaml` file? That way we need to changes to this package at all, and internally it should also be more straightforward.

Reply:
> I believe the pub authors don't like us modifying package_config.json after pub created that file.

Yeah, ideally it's best not to do this. It messes up timestamps. We can add support for it -- but I would suggest just writing it next to `package_config.json` in `.dart_tool/native_assets.yaml`.